### PR TITLE
CLI command to migrate legacy WP users

### LINKED ISSUE #78: Migration process for vanilla WordPress
Related: #23 

If this plugin is activated on an existing vanilla WordPress site (a site with no custom author attribution functionality) then it works but the `post_author` attribution is not used. This means author archives break.

A migration command needs to be added which copies the `post_author` field into a term relationship that Authorship uses.

### PR DESCRIPTION
**Pull request by mattheu**
If you enable Authorship on an existing site, author archives break because authorship authors are not set. 

This adds a CLI command to run through all posts that have no authorship author set, and sets adds the post_author to authorship.

Optionally, you can pass `--override-authorship` to remove all authorship users and replace with WordPress authors.

Fixes #78.

### CODE REVIEW

**Comment by svandragt** [comment]
I don't think the optional parameter is clear at all. How about `--set-wordpress-authors` to be descriptive of what the end result is?

File: inc/cli/class-migrate-command.php [hunk 1]
```diff
@@ -23,6 +23,119 @@
...
+				$authorship_authors = \Authorship\get_authors( $post );
+
+				if ( ! empty( $authorship_authors ) && ! $overwrite ) {
+					// skip posts that already have authorship data.
+					WP_CLI::line( 'Skipping post that already has Authorship data' );
+					continue;
+				}
+
+				// Set post author as Authorship author.
+				\Authorship\set_authors( $post, [ intval( $post->post_author ) ] );
```
Comment (johnbillion) [inline inc/cli/class-migrate-command.php:117]: This needs to be behind a check for the `dry-run` flag. See: https://github.com/humanmade/authorship/blob/108070304dcb8730f0a367c78e2cd16ebe2f1c7d/inc/cli/class-migrate-command.php#L275-L277

**Review by johnbillion — CHANGES_REQUESTED** [review]
This looks good, just needs:

1. A fix for the dry-run flag
2. Docs added to the readme (probably above the "PublishPress Authors Migration" section)

File: inc/cli/class-migrate-command.php (hunk 1 above)
Comment (mattheu) [reply inc/cli/class-migrate-command.php]: Oops. Somehow lost this change. I don't have a good local setup... copy/pasted from my project into a version of authorship in my project. Fixed.

**Comment by mattheu** [comment]
> I don't think the optional parameter is clear at all, but it's a great option to have. How about --set-wordpress-authors to be descriptive of what the end result is?

@svandragt I copied this command from the Publish Press Authors migration command, which does the exact same thing. So for consistency, I'd like to keep it the same.

**Comment by mattheu** [comment]
Feedback addressed @johnbillion

**Pull request merged**